From 37f99572d31b48e15a4a4b447114d2925de2ed74 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Wed, 2 Sep 2020 11:08:58 +0200
Subject: [PATCH] Add a debug print to load_library_permanently

.. for some reason Python 3.8 is failing on c3i on Windows with:

======================================================================
ERROR: test_imports (llvmlite.tests.test_llvmpy.TestMisc)
Sanity-check llvmpy APIs import correctly.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\ci\llvmlite_1599033872358\work\llvmlite\tests\test_llvmpy.py", line 15, in test_imports
    from llvmlite.llvmpy.core import Constant, Type, Builder  # noqa F401
  File "C:\ci\llvmlite_1599033872358\work\llvmlite\llvmpy\core.py", line 4, in <module>
    from llvmlite import binding as llvm
  File "C:\ci\llvmlite_1599033872358\work\llvmlite\binding\__init__.py", line 4, in <module>
    from .dylib import *
  File "C:\ci\llvmlite_1599033872358\work\llvmlite\binding\dylib.py", line 3, in <module>
    from llvmlite.binding import ffi
  File "C:\ci\llvmlite_1599033872358\work\llvmlite\binding\ffi.py", line 153, in <module>
    raise OSError("Could not load shared object file: {}".format(_lib_name))
OSError: Could not load shared object file: llvmlite.dll
----------------------------------------------------------------------
---
 llvmlite/binding/dylib.py | 17 +++++++++++++----
 llvmlite/binding/ffi.py   |  4 ++++
 2 files changed, 17 insertions(+), 4 deletions(-)

diff --git a/llvmlite/binding/dylib.py b/llvmlite/binding/dylib.py
index e22542c..d952629 100644
--- a/llvmlite/binding/dylib.py
+++ b/llvmlite/binding/dylib.py
@@ -24,10 +24,19 @@ def load_library_permanently(filename):
     """
     Load an external library
     """
-    with ffi.OutputString() as outerr:
-        if ffi.lib.LLVMPY_LoadLibraryPermanently(
-                _encode_string(filename), outerr):
-            raise RuntimeError(str(outerr))
+    print("load_library_permanently({})".format(filename))
+    from logging import getLogger
+    import os
+    log = getLogger(__name__)
+    log.warning("load_library_permanently({})".format(filename))
+    log.warning("load_library_permanently .. so.environ[PATH] is: {}".format(os.environ['PATH']))
+    try:
+        with ffi.OutputString() as outerr:
+            if ffi.lib.LLVMPY_LoadLibraryPermanently(
+                    _encode_string(filename), outerr):
+                raise RuntimeError(str(outerr)+'\nLIZZY PATH is: {}'.format(os.environ['PATH']))
+    except Exception as e:
+        raise RuntimeError(str(outerr)+'\nLIZZY2 PATH is: {}'.format(os.environ['PATH']))
 
 # ============================================================================
 # FFI
diff --git a/llvmlite/binding/ffi.py b/llvmlite/binding/ffi.py
index 3f53b2a..d990e29 100644
--- a/llvmlite/binding/ffi.py
+++ b/llvmlite/binding/ffi.py
@@ -140,9 +140,13 @@ except ImportError:
 else:
     _lib_paths.append(resource_filename(__name__, _lib_name))
 
+from logging import getLogger
+log = getLogger(__name__)
+log.warning("Load {} .. os.environ[PATH] is: {}".format(_lib_name, os.environ['PATH']))
 
 # Try to load from all of the different paths
 for _lib_path in _lib_paths:
+    log.warning("Checking {}".format(_lib_path))
     try:
         lib = ctypes.CDLL(_lib_path)
     except OSError:
-- 
2.24.3 (Apple Git-128)

